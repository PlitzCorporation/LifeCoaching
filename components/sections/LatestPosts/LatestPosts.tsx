import React from 'react';
import Link from 'next/link';

import { getPostsByCategory } from '@/helpers/pullBlogPost';

import BurgundyThickLine from '@/components/ui/decorations/BurgundyThickLine';
import ImagePlacer from '@/components/ui/ImagePlacer';
import SolidButton from '@/components/ui/SolidButton';

import { CategoryProps } from '@/types/posts';

const LatestPosts = async () => {
	const posts: CategoryProps = await getPostsByCategory('home-featured');

	const latestPosts = posts.items
		.sort((a, b) => Date.parse(b.createdAt) - Date.parse(a.createdAt))
		.slice(0, 3);

	return (
		<section>
			<div className="plitz-container flex flex-col items-center py-16 lg:py-28">
				<BurgundyThickLine />
				<div className="text-plitz-primary font-bold mt-7 lg:mt-10 mb-4">
					The Blog
				</div>
				<h3 className="mb-11 lg:mb-20">Latest Posts</h3>
				<div className="grid mb-9 lg:mb-12 grid-cols-1 gap-14 lg:grid-cols-3">
					{latestPosts.map((post) => (
						<Link
							href={`/blog/${post.link}`}
							key={post._id}
							className="flex flex-col items-center shadow-plitz-box-4-32 group"
						>
							<div className="w-full h-60">
								<ImagePlacer imageSrc={post.featuredimg} />
							</div>
							<div className=" px-6 lg:px-10 py-7 lg:py-9">
								<small className="text-plitz-primary">
									{new Date(post.createdAt).toLocaleDateString('en-US', {
										month: 'long',
										day: '2-digit',
										year: 'numeric',
									})}
								</small>
								<h4 className="text-2xl mt-4 mb-5 lg:text-3xl text-balance group-hover:text-plitz-primary transition-all group-hover:transition-all">
									{post.title}
								</h4>
								<div
									dangerouslySetInnerHTML={{ __html: post.postintro }}
									className="text-ellipsis line-clamp-3 text-plitz-body [&>p]:font-normal group-hover:text-plitz-primary"
								></div>
							</div>
						</Link>
					))}
				</div>
				<SolidButton label="View All Posts" href="/blog" />
			</div>
		</section>
	);
};

export default LatestPosts;
