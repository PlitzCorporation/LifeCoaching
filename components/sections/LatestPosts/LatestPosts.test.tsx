import { render } from '@testing-library/react';
import LatestPosts from './LatestPosts';

jest.mock('next/image', () => ({
	__esModule: true,
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	default: (props: any) => {
		// eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
		return <img {...props} priority="false" />;
	},
}));

jest.mock('@/helpers/apiCalls/pullBlogPost', () => ({
	getPostsByCategory: jest.fn().mockResolvedValue({
		items: [
			{
				_id: '1',
				link: 'post-1',
				title: 'Post 1',
				createdAt: '2021-09-01T00:00:00.000Z',
				featuredimg: 'image1.jpg',
				postintro: '<p>Post 1 intro</p>',
			},
			{
				_id: '2',
				link: 'post-2',
				title: 'Post 2',
				createdAt: '2021-09-02T00:00:00.000Z',
				featuredimg: 'image2.jpg',
				postintro: '<p>Post 2 intro</p>',
			},
			{
				_id: '3',
				link: 'post-3',
				title: 'Post 3',
				createdAt: '2021-09-03T00:00:00.000Z',
				featuredimg: 'image3.jpg',
				postintro: '<p>Post 3 intro</p>',
			},
		],
	}),
}));

describe('LatestPosts', () => {
	it('renders correctly with post data', async () => {
		const ComponentResolved = await LatestPosts();

		const MockComponent = () => ComponentResolved;
		const { container } = render(<MockComponent />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
