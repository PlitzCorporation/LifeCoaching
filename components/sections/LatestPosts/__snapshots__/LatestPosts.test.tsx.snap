// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatestPosts renders correctly with post data 1`] = `
<div>
  <section>
    <div
      class="plitz-container flex flex-col items-center py-16 lg:py-28"
    >
      <div
        class="block w-32 h-1 bg-gradient-to-r from-plitz-primary/5 to-plitz-primary/100"
      />
      <div
        class="text-plitz-primary font-bold mt-7 lg:mt-10 mb-4"
      >
        The Blog
      </div>
      <h3
        class="mb-11 lg:mb-20"
      >
        Latest Posts
      </h3>
      <div
        class="grid mb-9 lg:mb-12 grid-cols-1 gap-14 lg:grid-cols-3"
      >
        <a
          class="flex flex-col items-center shadow-plitz-box-4-32 group"
          href="/blog/post-3"
        >
          <div
            class="w-full h-60"
          >
            <img
              alt="image3"
              class="object-cover min-h-full min-w-full max-h-full max-w-full border-t-10 border-plitz-primary"
              height="1000"
              priority="false"
              src="undefined/image3.jpg"
              width="1000"
            />
          </div>
          <div
            class=" px-6 lg:px-10 py-7 lg:py-9"
          >
            <small
              class="text-plitz-primary"
            >
              September 02, 2021
            </small>
            <h4
              class="text-2xl mt-4 mb-5 lg:text-3xl text-balance group-hover:text-plitz-primary transition-all group-hover:transition-all"
            >
              Post 3
            </h4>
            <div
              class="text-ellipsis line-clamp-3 text-plitz-body [&>p]:font-normal group-hover:text-plitz-primary"
            >
              <p>
                Post 3 intro
              </p>
            </div>
          </div>
        </a>
        <a
          class="flex flex-col items-center shadow-plitz-box-4-32 group"
          href="/blog/post-2"
        >
          <div
            class="w-full h-60"
          >
            <img
              alt="image2"
              class="object-cover min-h-full min-w-full max-h-full max-w-full border-t-10 border-plitz-primary"
              height="1000"
              priority="false"
              src="undefined/image2.jpg"
              width="1000"
            />
          </div>
          <div
            class=" px-6 lg:px-10 py-7 lg:py-9"
          >
            <small
              class="text-plitz-primary"
            >
              September 01, 2021
            </small>
            <h4
              class="text-2xl mt-4 mb-5 lg:text-3xl text-balance group-hover:text-plitz-primary transition-all group-hover:transition-all"
            >
              Post 2
            </h4>
            <div
              class="text-ellipsis line-clamp-3 text-plitz-body [&>p]:font-normal group-hover:text-plitz-primary"
            >
              <p>
                Post 2 intro
              </p>
            </div>
          </div>
        </a>
        <a
          class="flex flex-col items-center shadow-plitz-box-4-32 group"
          href="/blog/post-1"
        >
          <div
            class="w-full h-60"
          >
            <img
              alt="image1"
              class="object-cover min-h-full min-w-full max-h-full max-w-full border-t-10 border-plitz-primary"
              height="1000"
              priority="false"
              src="undefined/image1.jpg"
              width="1000"
            />
          </div>
          <div
            class=" px-6 lg:px-10 py-7 lg:py-9"
          >
            <small
              class="text-plitz-primary"
            >
              August 31, 2021
            </small>
            <h4
              class="text-2xl mt-4 mb-5 lg:text-3xl text-balance group-hover:text-plitz-primary transition-all group-hover:transition-all"
            >
              Post 1
            </h4>
            <div
              class="text-ellipsis line-clamp-3 text-plitz-body [&>p]:font-normal group-hover:text-plitz-primary"
            >
              <p>
                Post 1 intro
              </p>
            </div>
          </div>
        </a>
      </div>
      <a
        class="relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl"
        data-testid="aTag"
        href="/blog"
      >
        <span
          class="translate-x-2 group-hover:-translate-x-1 transition-all"
        >
          View All Posts
        </span>
        <svg
          class="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
          height="1em"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
          />
        </svg>
      </a>
    </div>
  </section>
</div>
`;
